swift/apple/FirezoneNetworkExtension/SessionManager.swift

```diff
@@ -0,0 +1,250 @@
+actor SessionManager {
+  // Session ownership
+  private var session: Session?
+  private var eventLoopTask: Task<Void, Never>?
+
+  // Command channel (Swift equivalent of Kotlin's Channel)
+  private let commandContinuation: AsyncStream<SessionCommand>.Continuation
+  private let commandStream: AsyncStream<SessionCommand>
+
+  // Event handler callback
+  private let eventHandler: (Event) async -> Void
+
+  // Logging interval for event loop health monitoring
+  private static let eventPollingLogInterval = 100
+
+  init(eventHandler: @escaping (Event) async -> Void) {
+    self.eventHandler = eventHandler
+    (commandStream, commandContinuation) = AsyncStream.makeStream()
+  }
+
+  /// Starts the session and event loop.
+  ///
+  /// This creates a Rust Session via UniFFI and spawns the event loop task
+  /// that owns the session lifecycle.
+  func start(
+    apiUrl: String,
+    token: Token,
+    deviceId: String,
+    accountSlug: String,
+    logFilter: String
+  ) throws {
+    Log.log("SessionManager: Starting session for account: \(accountSlug)")
+
+    // Get device metadata
+    let deviceName = DeviceMetadata.getDeviceName()
+    let osVersion = DeviceMetadata.getOSVersion()
+    let deviceInfo = try JSONEncoder().encode(DeviceMetadata.deviceInfo())
+    let deviceInfoStr = String(data: deviceInfo, encoding: .utf8) ?? "{}"
+    let logDir = SharedAccess.connlibLogFolderURL?.path ?? "/tmp/firezone"
+
+    // Create the session
+    let session: Session
+    do {
+      session = try Session.newApple(
+        apiUrl: apiUrl,
+        token: token.description,
+        deviceId: deviceId,
+        accountSlug: accountSlug,
+        deviceName: deviceName,
+        osVersion: osVersion,
+        logDir: logDir,
+        logFilter: logFilter,
+        deviceInfo: deviceInfoStr
+      )
+    } catch {
+      Log.error(error)
+      throw AdapterError.connlibConnectError(String(describing: error))
+    }
+
+    self.session = session
+
+    // Start event loop - THIS OWNS THE SESSION
+    eventLoopTask = Task { [weak self] in
+      await self?.runEventLoop(session: session)
+    }
+
+    Log.log("SessionManager: Session started successfully")
+  }
+
+  /// The main event loop that owns the Session lifecycle.
+  ///
+  /// This uses withTaskGroup to multiplex between:
+  /// 1. Event polling from Rust (nextEvent)
+  /// 2. Command handling from Swift (command channel)
+  ///
+  /// When either task completes, both are cancelled and the session is cleaned up.
+  /// This ensures the Session's Drop is called on the Rust side.
+  private func runEventLoop(session: Session) async {
+    Log.log("SessionManager: Starting event loop")
+
+    // Multiplex between commands and events
+    await withTaskGroup(of: Void.self) { group in
+      // Event polling task - polls Rust for events
+      group.addTask { [weak self] in
+        guard let self = self else { return }
+
+        var eventCount = 0
+        var pollAttempts = 0
```

Comment:
Is the purpose of this just logging? If so, I think we can remove it to keep things simpler.